src/components/Dropdown/Dropdown.tsx

```diff
@@ -0,0 +1,92 @@
+import { useCallback, useState } from 'react';
+import { Person } from '../../types/Person';
+import React from 'react';
+import classNames from 'classnames';
+import debounce from 'lodash.debounce';
+import { peopleFromServer } from '../../data/people';
+
+type Props = {
+  onSelected: (person: Person) => void;
+  delay: number;
+  inputIsFocused: boolean;
+  filteredList: Person[];
+  appliedQuery: string;
+  setAppliedQuery: (arg: string) => void;
+  setSelectedPerson: (person?: Person) => void;
+};
+
+export const Dropdown: React.FC<Props> = React.memo(
+  ({
+    onSelected,
+    delay,
+    filteredList,
+    setAppliedQuery,
+    appliedQuery,
+    setSelectedPerson,
+  }) => {
+    const [query, setQuery] = useState('');
+    const [showSuggestions, setShowSuggestions] = useState(false);
+
+    const applyQuery = useCallback(debounce(setAppliedQuery, delay), []);
```

Comment:
pls fix this linter issue

the reason here is when you pass function to usecallback React cannot define its dependencies. You need to pass inline function like this:

```suggestion
    const applyQuery = useCallback(() => debounce(setAppliedQuery, delay), []);
```